### DISCUSSION TITLE
How can we add packages / extensions to global environments

### DISCUSSION THREAD

**Discussion by jslorrma** [prefix-dev/pixi#655] [Q&A] [answered]
First of all, I'm really hyped about pixi as a (python) project managing tool. 

In our company and on our working laptops the default python distribution is `Anaconda` together with internal mirror-servers for conda channels. So conda is usually and also the recommended python eco-system used. Anyhow user-experience with conda virtual environments, especially using conda tool, can be very disappointing and exhausting ...

First of all pixi is great because it's super fast and secondly it uses many design concepts and ideas I have used and implemented in a multi-platform toolchain ([Taskfile](https://github.com/go-task/task), micromamba, pyproject.toml, boa, some bash-, powershell- and python-scripts, [copier](https://github.com/copier-org/copier) + some CI stuff) to standardize, automate and manage (python) library and application (mainly data analysis & science) projects (fyi: [library vs. application)](https://pdm-project.org/latest/usage/project/#library-or-application) in our department. I strongly believe that in the future, with pixi, this toolchain can be very much simplified, especially regarding the massive struggle making the toolchain platform independent and it's nice pypi integration. My first tests and trials with pixi give me a lot of optimism ... 

So here comes my question: 
One concept we are using in our toolchain is the one of global installs, for instance for our CI tooling. But we also use a global `jupyter` installation (basically a jupyter conda environment) together with `ipykernel` within the individual projects (projects use local conda environments). 
If I use pixi to globally install `jupyter` how can I install additional packages, like e.g. [jupytext](https://github.com/mwouts/jupytext), or notebook- / lab-extensions like [jupyterlab-git](https://github.com/jupyterlab/jupyterlab-git) into the global jupyter environment?

**Comment by ruben-arts** [comment, marked as answer]
He @jslorrma

Cool to hear your experience and thoughts. I think you describe the exact situation we had that triggered the development of `pixi`. 

I guess what you're looking for is something previously described as `pixi global inject`, see this issue: https://github.com/prefix-dev/pixi/issues/342

On another note, we are currently looking for companies that want to be design partners on the work in and around pixi, seeing your enthusiasm and current workflow this might be awesome for the both of us! 
If you're interested in a call, contact us through:  hi@prefix.dev.

**Reply by jslorrma** [reply]
Hi @ruben-arts,

thanks for your quick reply.  Any plans yet when you are working on the `pixi global inject `feature or any idea when the feature will be available?

In case you don't have it on your roadmap yet, this might be an opportunity for me to contribute, if it is kind of straight forward and and mainly about the implementation and less about the general concept and design. I have senior (~20 year) experience in SW development (C, C++, Python, ...) but only basic knowledge developing in rust - I've done a few toy projects to get to know the language and its concepts. But I can give it a try. What do you think?

> If you're interested in a call, contact us through: [hi@prefix.dev](mailto:hi@prefix.dev).

I'll be back at work next week and will discuss this with my colleagues (at first I have to introduce them to pixi and get them hyped, too 😉)

**Reply by ruben-arts** [reply]
Our core dev team is currently not focusing on `pixi global` there have been some nice additions by the community. We're welcoming any PR's and might help with tips on rust development. If you want to discuss implementation details it best to do that in the issue #342. 

Sure, just ping us if you want to have a introductory call!

**Comment by lucascolley** [comment]
https://pixi.sh/latest/global_tools/manifest/#the-global-manifest
